File: app/src/main/java/com/ubiqsmart/ui/widgets/charts/HourXFormatter.java
package com.ubiqsmart.ui.widgets.charts;

import com.github.mikephil.charting.components.AxisBase;
import com.github.mikephil.charting.formatter.IAxisValueFormatter;

import java.util.*;

public class HourXFormatter implements IAxisValueFormatter {

  private GregorianCalendar c = new GregorianCalendar();

  @Override public String getFormattedValue(float value, AxisBase axis) {
    c.setTimeInMillis(((long) (value)) * 1000);
    return c.get(Calendar.HOUR_OF_DAY) + ":00 ";
  }
}

File: app/src/main/java/com/ubiqsmart/app/ui/widgets/charts/HourXFormatter.kt
package com.ubiqsmart.app.ui.widgets.charts

import com.github.mikephil.charting.components.AxisBase
import com.github.mikephil.charting.formatter.IAxisValueFormatter

import java.util.*

class HourXFormatter : IAxisValueFormatter {

  private val c = GregorianCalendar()

  override fun getFormattedValue(value: Float, axis: AxisBase): String {
    c.timeInMillis = value.toLong() * 1000
    return c.get(Calendar.HOUR_OF_DAY).toString() + ":00 "
  }
}
